### PR TITLE
Bump libcalico go

### PR DESCRIPTION
**Pull request by ozdanborne**
## Description

- Bumps libcalico-go to latest version
- fixes some sample policies that violated new icmp validation

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: tests/st/utils/data.py [hunk 1]
```diff
@@ -260,7 +258,6 @@
         ],
         'ingress': [
             {
-                'notICMP': {'type': 19, 'code': 255},
```
Comment (caseydavenport) [inline tests/st/utils/data.py]: Why are we removing these? Which part of validation does this fail?

Comment (matthewdupre) [reply tests/st/utils/data.py]: There's a proto: TCP just below.

Comment (ozdanborne) [reply tests/st/utils/data.py]: These rules have both ICMP rules and Ports rules, which is invalid now:
- If includes port rules, protocol needs to be TCP. 
- If it includes ICMP rules, protocol needs to be ICMP

**Comment by caseydavenport** [comment]
@ozdanborne could you write a release note?

**Pull request merged**